lib/iris/mesh/components.py

```diff
@@ -2687,12 +2767,15 @@ class MeshCoord(AuxCoord):
 
     def __init__(
         self,
-        mesh: MeshXY,
-        location: Location,
-        axis: Literal["x", "y"],
+        mesh,
+        location,
+        axis,
     ):
+        self.timestamp = None
```

Comment:
This should probably be private. It might also benefit froma different name to identify that this has a different role to the other variables named `timestamp`. Maybe something like `_last_recorded_mesh_update`.

Reply:
This isn't to hold when the mesh was last updated, but rather when the MeshCoord was last updated. This is then compared against the mesh one. 

As such, I think this *should* have the same name as on the mesh, (I'm chaging both to `_last_modified`), because then the line:

`if self._last_modified != ...mesh._last_modified:` is more clear. Is this okay?